app/classes/Transvision/Search.php

```diff
@@ -0,0 +1,228 @@
+<?php
+namespace Transvision;
+
+/**
+ * Search class
+ *
+ * Allows searching for data in our repositories using a fluent interface.
+ * Currently, only the regex part (definition of the search) is implemented.
+ * ex:
+ * $search = (new Search)
+ *     ->setSearchTerms('Bookmark this page')
+ *     ->setRegexWholeWords(true)
+ *     ->setRegexCase('sensitive')
+ *     ->setRegexPerfectMatch(false);
+ */
+class Search
+{
+    /**
+     * The trimmed string searched, we keep that one as the canonical reference
+     * @var string
+     */
+    protected $search_terms;
+
+    /**
+     * The generated regex string updated dynamically via updateRegex()
+     * @var string
+     */
+    protected $regex;
+
+    /**
+     * Case sensibility of the regex
+     * @var string
+     */
+    protected $regex_case;
+
+    /**
+     * Consider the space separated string as a single word for search
+     * @var string
+     */
+    protected $regex_whole_words;
+
+    /**
+     * Only return strings that match the search perfectly (case excluded)
+     * @var boolean
+     */
+    protected $regex_perfect_match;
+
+    /**
+     * The search terms for the regex, those differ from $search_terms as
+     * they can be changed dynamically via setRegexSearchTerms()
+     * @var string
+     */
+    protected $regex_search_terms;
+
+    /**
+     * We set the default values for a search
+     */
+    public function __construct()
+    {
+        $this->search_terms = '';
+        $this->regex = '';
+        $this->regex_case = 'i';
+        $this->regex_whole_words = '';
```

Comment:
that means making updateRegex() more complex as we need then to change them back to strings there